release.sh

```diff
@@ -301,6 +302,31 @@ function build_from_source() {
   if [[ $? -ne 0 ]]; then
     abort "error building the release"
   fi
+  sign_release
+  # Do not use `||` above or any error will be swallowed.
+  if [[ $? -ne 0 ]]; then
+    abort "error signing the release"
+  fi
+}
+
+# Build a release from source.
+function sign_release() {
+  ## Sign the images with cosign
+  ## For now, check if ko has created imagerefs.txt file. In the future, missing image refs will break
+  ## the release for all jobs that publish images.
+  if [[ -f "imagerefs.txt"  && -z ${SIGNING_KEY} ]]; then
+      echo "Signing Images with the key ${SIGNING_KEY}"
```

Comment:
It looks like this means we'll be signing the images but not the manifests that reference the images. Should we be signing those as well?

Reply:
I don't think it is needed. Endusers are most like going to do something like this: https://kyverno.io/docs/writing-policies/verify-images/ so only the image should be signed for now.

Reply:
I agree but in the kubernetes images, we chose to sign everything that gets tagged. We sign the index as there are situations (say in a multiarch cluster) where you may choose to do the verification on the index instead of a single image. Another reason is that at some point, you may want to attach sboms to the images so it makes sense to have the index signed to validate the index sbom too.

(Hint `cosign sign --recursive` is your friend here)